src/crypto/test/CryptoTests.cpp

```diff
@@ -451,3 +452,49 @@ TEST_CASE("StrKey tests", "[crypto]")
     LOG_INFO(DEFAULT_LOG, "CRC16 error-detection rate {}", detectionRate);
     REQUIRE(detectionRate > 99.99);
 }
+
+TEST_CASE("key string roundtrip", "[crypto]")
+{
+    SignerKey signer;
+    auto publicKey = SecretKey::random().getPublicKey();
+    uint256 rand256 = publicKey.ed25519();
+    SECTION("SIGNER_KEY_TYPE_ED25519")
+    {
+        signer.type(SIGNER_KEY_TYPE_ED25519);
+        signer.ed25519() = rand256;
+        REQUIRE(KeyUtils::fromStrKey<SignerKey>(KeyUtils::toStrKey(signer)) ==
+                signer);
+    }
+    SECTION("SIGNER_KEY_TYPE_PRE_AUTH_TX")
+    {
+        signer.type(SIGNER_KEY_TYPE_PRE_AUTH_TX);
+        signer.preAuthTx() = rand256;
+        REQUIRE(KeyUtils::fromStrKey<SignerKey>(KeyUtils::toStrKey(signer)) ==
+                signer);
+    }
+    SECTION("SIGNER_KEY_TYPE_HASH_X")
+    {
+        signer.type(SIGNER_KEY_TYPE_HASH_X);
+        signer.hashX() = rand256;
+        REQUIRE(KeyUtils::fromStrKey<SignerKey>(KeyUtils::toStrKey(signer)) ==
+                signer);
+    }
+    SECTION("SIGNER_KEY_TYPE_ED25519_SIGNED_PAYLOAD")
+    {
+        signer.type(SIGNER_KEY_TYPE_ED25519_SIGNED_PAYLOAD);
+        signer.ed25519SignedPayload().ed25519 = rand256;
+
+        for (int i = 0; i < signer.ed25519SignedPayload().payload.max_size();
```

Comment:
i should be `uint32_t` (warning, same type than `max_size()`)

Reply:
fixed